[Refactor](runtime-filter) Refactor code about runtime filter

BiteTheDDDDt opened the pull request:
### What problem does this PR solve?
0. move code about runtime filter to single root directory 'runtime_filter'
1. split runtime filter class to producer/merger/consumer, and add some state system to represent rf's progress
2. add FilterOlapParam to pass filter id to storage later
3. add some beut of runtime filter
4. some others refactor
5. add some inject test

### Check List (For Author)

- Test <!-- At least one of them must be included. -->
    - [x] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason <!-- Add your reason?  -->

- Behavior changed:
    - [x] No.
    - [ ] Yes. <!-- Explain the behavior change -->

- Does this need documentation?
    - [x] No.
    - [ ] Yes. <!-- Add document PR link here. eg: https://github.com/apache/doris-website/pull/1214 -->

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label <!-- Add branch pick label that this PR should merge into -->

Commit:
refactor of RuntimeFilterSlots

refactor of RuntimeFilter

update

update

update

update

update

update

fix

fix

rename consumer

update

fix

update

fix

move directory

add runtime filter helper

update role state system

fix producer sstate

update consumer state

fix

update

update

fix

update

update

format

update

update

update

update profile

format

fix

fix

update

format

update

fix

update

update

update

update

[refactor](runtime filter) Simplify producer / slots (#48145)

[refactor](runtime filter) Refine consumer (#48198)

update filters refactor

fix

update

update

update

move

update

update

update

update

update

remove fuzzy

update some fix

update fix

update fix

update fix

fix rebase

fix ut

fix

update

update fix

update

update

update

hello-stephen commented:
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

BiteTheDDDDt commented:
run  buildall

Commit:
some fix and add ut

fix ut

update

update

format

udpate

BiteTheDDDDt commented:
run buildall

Gabriel39 committed:
[refactor](runtime filter) Refine null property in runtime filter (#48534)

### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test <!-- At least one of them must be included. -->
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
- [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason <!-- Add your reason?  -->

- Behavior changed:
    - [ ] No.
    - [ ] Yes. <!-- Explain the behavior change -->

- Does this need documentation?
    - [ ] No.
- [ ] Yes. <!-- Add document PR link here. eg:
https://github.com/apache/doris-website/pull/1214 -->

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label <!-- Add branch pick label that this PR
should merge into -->

BiteTheDDDDt commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `20.21% (807/3993)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/2a85ea0d89d68d7c9a7d8d7e716ead92a0ccd0d8_2a85ea0d89d68d7c9a7d8d7e716ead92a0ccd0d8/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/2a85ea0d89d68d7c9a7d8d7e716ead92a0ccd0d8_2a85ea0d89d68d7c9a7d8d7e716ead92a0ccd0d8/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 45.39% (12100/26657) |
| Line Coverage     | 34.81% (101716/292236) |
| Region Coverage   | 33.96% (52040/153256) |
| Branch Coverage   | 29.80% (26433/88688) |

Gabriel39 committed:
[runtime filter](UT) test in filter (#48562)

Commit:
add more test and some fix

BiteTheDDDDt commented:
run buildall

Gabriel39 committed:
[runtime filter](UT) test bloom filter (#48575)

Commit:
fix coredump

BiteTheDDDDt commented:
run buildall

Gabriel39 committed:
[rf](ut) test minmax/bitmap filter (#48579)

Commit:
fix compile fail and format

Gabriel39 committed:
fix UT (#48591)

Gabriel39 commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `37.29% (1464/3926)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/2292611437b3a18040dd769520a97041071790f7_2292611437b3a18040dd769520a97041071790f7/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/2292611437b3a18040dd769520a97041071790f7_2292611437b3a18040dd769520a97041071790f7/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 46.23% (12336/26684) |
| Line Coverage     | 35.63% (104193/292390) |
| Region Coverage   | 34.76% (53318/153379) |
| Branch Coverage   | 30.47% (27020/88682) |

Gabriel39 committed:
test in or bloom (#48596)

Commit:
add more ut of roles

BiteTheDDDDt commented:
run buildall

Gabriel39 committed:
test rf utils (#48615)

Gabriel39 committed:
test (#48630)

Gabriel39 committed:
format (#48658)

Gabriel39 committed:
test RuntimeFilterMgr (#48672)

Gabriel39 committed:
test RuntimeFilterMergeControllerEntity (#48678)

Gabriel39 commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `43.97% (1726/3925)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/8566f1463cecf0dc847002c74730afd3f2a39956_8566f1463cecf0dc847002c74730afd3f2a39956/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/8566f1463cecf0dc847002c74730afd3f2a39956_8566f1463cecf0dc847002c74730afd3f2a39956/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 46.78% (12487/26691) |
| Line Coverage     | 36.16% (105764/292461) |
| Region Coverage   | 35.27% (54112/153433) |
| Branch Coverage   | 31.02% (27513/88708) |

Commit:
add rpc fail test and eof test

update

update fix

fix case

fix case

fix case

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `43.83% (1725/3936)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/a65b45785cb9e618e016e751c4a10d62e0c5574c_a65b45785cb9e618e016e751c4a10d62e0c5574c/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/a65b45785cb9e618e016e751c4a10d62e0c5574c_a65b45785cb9e618e016e751c4a10d62e0c5574c/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 46.80% (12494/26696) |
| Line Coverage     | 36.18% (105855/292561) |
| Region Coverage   | 35.28% (54140/153467) |
| Branch Coverage   | 31.03% (27534/88732) |

BiteTheDDDDt commented:
run buildall

mrhhsg committed:
[test](ut) add ut for bloom filter and hybridset (#48698)

### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test <!-- At least one of them must be included. -->
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
- [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason <!-- Add your reason?  -->

- Behavior changed:
    - [ ] No.
    - [ ] Yes. <!-- Explain the behavior change -->

- Does this need documentation?
    - [ ] No.
- [ ] Yes. <!-- Add document PR link here. eg:
https://github.com/apache/doris-website/pull/1214 -->

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label <!-- Add branch pick label that this PR
should merge into -->

BiteTheDDDDt commented:
run buildall

mrhhsg committed:
[test](bloom_filter_func) add case about algorithm

BiteTheDDDDt commented:
run buildall

Commit:
resolve conflict

BiteTheDDDDt commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `47.23% (1869/3957)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/119abb6e6cd3b3d35de48e252fafc91c3a6323ff_119abb6e6cd3b3d35de48e252fafc91c3a6323ff/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/119abb6e6cd3b3d35de48e252fafc91c3a6323ff_119abb6e6cd3b3d35de48e252fafc91c3a6323ff/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 47.98% (12818/26713) |
| Line Coverage     | 37.43% (109561/292744) |
| Region Coverage   | 36.38% (55871/153581) |
| Branch Coverage   | 31.73% (28180/88812) |

mrhhsg committed:
[test](rf) Improve test about bloom filter func

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run beut

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

Commit:
add some metrics

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

mrhhsg committed:
[test](ut) Increase the types coverage of BloomFilterFunc

BiteTheDDDDt commented:
run buildall

Commit:
use has_remote_target to represent mode and add bitmap filter test

BiteTheDDDDt commented:
run buildall

mrhhsg committed:
[test](ut) [test](ut) Increase the types coverage of HybridSet/MinMax

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

Commit:
add lock for producer set state

BiteTheDDDDt commented:
run buildall

Commit:
fix ut compile

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

Commit:
fix DummyWorkloadGroupTest

BiteTheDDDDt commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `80.14% (1921/2397)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/b1e99309aee651c77b9572c9300f555349d3571a_b1e99309aee651c77b9572c9300f555349d3571a/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/b1e99309aee651c77b9572c9300f555349d3571a_b1e99309aee651c77b9572c9300f555349d3571a/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 49.94% (13360/26753) |
| Line Coverage     | 39.38% (115515/293347) |
| Region Coverage   | 38.09% (58647/153965) |
| Branch Coverage   | 33.23% (29601/89078) |

Commit:
add compile check

BiteTheDDDDt commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `80.14% (1921/2397)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/6f05d0408b9575275c6cf62816e3e4b145761c33_6f05d0408b9575275c6cf62816e3e4b145761c33/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/6f05d0408b9575275c6cf62816e3e4b145761c33_6f05d0408b9575275c6cf62816e3e4b145761c33/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 49.96% (13365/26753) |
| Line Coverage     | 39.39% (115539/293351) |
| Region Coverage   | 38.10% (58662/153968) |
| Branch Coverage   | 33.24% (29607/89080) |

BiteTheDDDDt commented:
run buildall

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `80.14% (1921/2397)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/6f05d0408b9575275c6cf62816e3e4b145761c33_6f05d0408b9575275c6cf62816e3e4b145761c33/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/6f05d0408b9575275c6cf62816e3e4b145761c33_6f05d0408b9575275c6cf62816e3e4b145761c33/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 49.94% (13360/26750) |
| Line Coverage     | 39.37% (115477/293309) |
| Region Coverage   | 38.09% (58638/153958) |
| Branch Coverage   | 33.21% (29583/89074) |
